pkg/probeupload/names_test.go

```diff
@@ -35,6 +37,8 @@ func TestIsValidModuleVersion_Invalid(t *testing.T) {
 		".2.3",
 		"2.3.",
 		"10.5.155.123",
+		"1.0.0-rc",
+		"2.0.0-invalid",
```

Comment:
Pretty silly nitpick, but could we add a few cases like `1.0.0-rc1-rc4` and `2.0.0-rc1invalid`?